cleanlab/internal/multilabel_rank_utils.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Helper classes and functions used internally for computing multilabel label quality scores.
+"""
+
+from enum import Enum
+from typing import Callable, Optional
+
+import numpy as np
+from sklearn.utils.multiclass import is_multilabel
+
+from cleanlab.rank import (
+    get_self_confidence_for_each_label,
+    get_normalized_margin_for_each_label,
+    get_confidence_weighted_entropy_for_each_label,
+)
+
+
+class _Wrapper:
+    """Helper class for wrapping callable functions as attributes of an Enum instead of
+    setting them as methods of the Enum class.
+
+
+    This class is only intended to be used internally for the BaseQualityScorer or
+    other cases where functions are used for enumeration values.
+    """
+
+    def __init__(self, f: Callable) -> None:
+        self.f = f
+
+    def __call__(self, *args, **kwargs):
+        return self.f(*args, **kwargs)
+
+    def __repr__(self):
+        return self.f.__name__
+
+
+class BaseQualityScorer(Enum):
```

Comment:
maybe call this something more specific like: `ClassLabelScorer` or `ClassificationLabelScorer` ?  The name is currently a bit vague, especially now that we're starting to branch beyond standard multiclass classification tasks, as well as scoring other issues beyond label errors.

Reply:
```suggestion
class ClassLabelScorer(Enum):
```